Mathlib/MeasureTheory/Measure/Real.lean

```diff
@@ -0,0 +1,434 @@
+/-
+Copyright (c) 2025 Sébastien Gouëzel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel
+-/
+import Mathlib.MeasureTheory.Measure.Prod
+import Mathlib.Tactic.Finiteness
+
+/-!
+# Measures as real valued-functions
+
+Given a measure `μ`, we define `μ.real` as the function sending a set `s` to `(μ s).toReal`, and
+develop a basic API around this.
+
+We essentially copy relevant lemmas from the files `MeasureSpaceDef.lean`, `NullMeasurable.lean` and
+`MeasureSpace.lean`, and adapt them by replacing in their name `measure` with `measureReal`.
+
+Many lemmas require an assumption that some set has finite measure. These assumptions are written
+in the form `(h : μ s ≠ ∞ := by finiteness)`, where `finiteness` is a tactic for goals of
+the form `≠ ∞`.
+
+There are certainly many missing lemmas. The missing ones should be added as they are needed.
+
+There are no lemmas on infinite sums, as summability issues are really
+more painful with reals than nonnegative extended reals. They should probably be added in the long
+run.
+-/
+
+open MeasureTheory Measure Set
+open scoped ENNReal NNReal Function symmDiff
+
+namespace MeasureTheory
+
+variable {α β ι : Type*} {_ : MeasurableSpace α} (μ : Measure α) {s s₁ s₂ s₃ t t₁ t₂ u : Set α}
+
+/-- The real-valued version of a measure. Maps infinite measure sets to zero. Use as `μ.real s`. -/
+protected def Measure.real (s : Set α) : ℝ :=
+  (μ s).toReal
+
+theorem measureReal_def (s : Set α) : μ.real s = (μ s).toReal := rfl
+
+/-- The nnreal-valued version of a measure. Maps infinite measure sets to zero.
+Use as `μ.nnreal s`. -/
+protected def Measure.nnreal (s : Set α) : ℝ≥0 :=
+  (μ s).toNNReal
+
+theorem measureNNReal_def (s : Set α) : μ.nnreal s = (μ s).toNNReal := rfl
+
+@[simp] theorem measureNNReal_toReal (s : Set α) : (μ.nnreal s).toReal = μ.real s := rfl
+
+theorem measureNNReal_val (s : Set α) : (μ.nnreal s).val = μ.real s := rfl
```

Comment:
You define that nnreal version but there is no API about it beyond this point. Let's delete it?
Also, working with nnreal is often not more pleasant than ennreal anyway, so I am not sure that it's really useful.